### PR TITLE
HOTFIX: Fix sequence templates time helper functions

### PR DESCRIPTION
**Pull request by pranav-super**
`___REQUIRES_GATEWAY_PR___="132"`

* **Tickets addressed:** N/A
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
There are a few minor fixes required in `InstantToSTOL`/`InstantToText` conversion. These primarily have to do with the presence of milliseconds versus microseconds, and the presence of a Zulu time indicator being required versus not. This gets its own PR, separate from the Sequence Template work, because the functionality here _does_ affect other parts of the code and isn't exclusively related to Sequence Templates. That being said, the changes are pretty minimal.

## Verification
All tests are updated! It's one of the 4 commits in this hotfix.

## Documentation
NASA-AMMOS/aerie-docs#243

## Future work
N/A

### CODE REVIEW

File: sequencing-server/src/lib/mustache/util/time.ts [hunk 1]
```diff
@@ -180,7 +177,16 @@ export function InstanttoSeqN(date: Temporal.Instant): string { // presently, ca
...
-  return result + (result.includes("Z") ? "" : "Z");
+  if (result.includes(".") && result.split(".").length == 2) {
+    let split = result.split(".")
+    let rest = split[0]
+    let micros = split[1]?.replace("Z", "")
+    micros = micros?.padEnd(6, "0")
+    return rest + "." + micros + "Z"
+  }
+  else {
+    return result + (result.includes("Z") ? "" : "Z");
```
Comment (cartermak) [inline sequencing-server/src/lib/mustache/util/time.ts]:
```suggestion
    return rest + "." + micros
  }
  else {
    return result + "Z";
```

**Review by cartermak — COMMENTED** [review]
I opened #1703 with the original goal of just getting `format-as-date` caught up with `add-time` and `subtract-time` changes here, but it became a slightly bigger set of changes. I could support merging this as-is and merging that separately, or combining them here (I opened that PR as merging into this branch).

File: sequencing-server/src/lib/mustache/util/time.ts (hunk 1 above)
Comment (cartermak) [reply sequencing-server/src/lib/mustache/util/time.ts]: idk what I was going for here but I think it's fine...

File: sequencing-server/test/command-expansion.spec.ts [hunk 2]
```diff
@@ -1229,5 +1229,89 @@ CMD SEQUENCE=FINAL_B AT=2020-001/00:00:30.030Z`) // expect interleaving!
...
+      // remove sequence
+      await removeSequence(graphqlClient, seqId);
+      // remove simulation artifact pk
+      await removeSimulationArtifacts(graphqlClient, simulationArtifactPk);
+      // remove associations
+      await removeActivitySequenceAssignments(graphqlClient, seqId);
+    });
+  });
+
   // TODO: DAVID'S EDGE CASES
```
Comment (cartermak) [inline sequencing-server/test/command-expansion.spec.ts]: Hmmmm...I wonder what David's edge cases are?

**Review by cartermak — APPROVED** [review]
I'm a little curious why the preferred behavior is truncation rather than rounding for the STOL timestamps but if that's the intended behavior, the implementation looks good

File: sequencing-server/test/command-expansion.spec.ts (hunk 2 above)
Comment (pranav-super) [reply sequencing-server/test/command-expansion.spec.ts]: The messages are more than 90 days old so I'm not sure...I can go ahead and remove this comment because it doesn't really add anything?

**Review by pranav-super — COMMENTED** [review]
Seems fine, just need to rebase the commits!

File: sequencing-server/test/command-expansion.spec.ts (hunk 2 above)
Comment (cartermak) [reply sequencing-server/test/command-expansion.spec.ts]: Yeah have at it
